chain/client/src/view_client_actor.rs

```diff
@@ -152,7 +153,7 @@ impl ViewClientActorInner {
                 runtime: runtime.clone(),
                 network_adapter: network_adapter.clone(),
                 config: config.clone(),
-                request_manager: Arc::new(RwLock::new(ViewClientRequestManager::new())),
+                request_manager: request_manager.clone(),
```

Comment:
This is a great find! 

Was this actually breaking the test or is it unrelated perf improvement?

Reply:
yea it actually breaks the test because the `send_tx_and_wait()` call in the test calls `broadcast_tx_commit`, which ends up hitting this logic when the rpc handler loops and polls for the tx status. that rpc call was timing out even tho the tx was successful because the thread that handled the `broadcast_tx_commit` rpc messages was different than the one that received the tx status response from the validator